### ISSUE TITLE
Typescript & Javascript language features extension in VSCode causes my laptop to overheat

### ISSUE THREAD

**Issue by MohamedWagih96** [microsoft/vscode#90980]
- VSCode Version: 1.42.1
- OS Version: Ubuntu 18.04.4 LTS

Steps to Reproduce:

1. Launch VSCode.
2. Enable Typescript and Javascript language features extension.

Does this issue occur when all extensions are disabled?: 

No, As a matter of fact I disabled all the built-in extensions except **Typescript and Javascript language features** extension and the problem remains untacked but if I disabled it no more loud fans nor overheat.
Also, I tried the` code --status` command mentioned [here](https://github.com/microsoft/vscode/wiki/Performance-Issues) and I found that extensionHost is listed in the output and as they stated 

> When the process listing in the Process Explorer or the output of code --status shows that the extensionHost is consuming a high percentage of CPU then this is likely caused by an extension.

Finally, I monitored the CPU usage % while running VSCode with and without the mentioned extension I found the following:

- When the extension is disabled: 0%
- When the extension is enabled: Starts at 30% to 28% to 16% then keeps bouncing between 16% and 9%.

This issue overheats my laptop and I think that was not intended I mean why only this extension is causing me issues while the others are not? and no I can't just disable it and move on because I need it. The whole point of me using VSCode is this feature.

*P.S: I tried clean install and disabled gpu acceleration nothing worked.

**Comment by mjbvz** [comment]
Can you share the output of `code --status`

Also, does this reproduce in the [latest VS Code insiders build](https://code.visualstudio.com/insiders/)?

**Comment by MohamedWagih96** [comment]
> Can you share the output of code --status

<pre>Version:          Code 1.42.1 (c47d83b293181d9be64f27ff093689e8e7aed054, 2020-02-11T14:50:36.977Z)
OS Version:       Linux x64 4.15.0-88-generic
CPUs:             Intel(R) Core(TM) i7-5500U CPU @ 2.40GHz (4 x 2770)
Memory (System):  7.71GB (2.04GB free)
Load (avg):       3, 2, 1
VM:               0%
Screen Reader:    no
Process Argv:     --disable-gpu --unity-launch
GPU Status:       2d_canvas:                     unavailable_software
                  flash_3d:                      disabled_software
                  flash_stage3d:                 disabled_software
                  flash_stage3d_baseline:        disabled_software
                  gpu_compositing:               disabled_software
                  metal:                         disabled_off
                  multiple_raster_threads:       enabled_on
                  oop_rasterization:             disabled_off
                  protected_video_decode:        disabled_off
                  rasterization:                 disabled_software
                  skia_renderer:                 disabled_off
                  surface_control:               disabled_off
                  surface_synchronization:       enabled_on
                  video_decode:                  disabled_software
                  viz_display_compositor:        enabled_on
                  viz_hit_test_surface_layer:    disabled_off
                  webgl:                         unavailable_software
                  webgl2:                        unavailable_software

CPU %	Mem MB	   PID	Process
    0	   134	  9036	code main
    0	    32	  9040	   zygote
    0	     8	  9042	     zygote
    0	    79	  9066	   gpu-process
    0	     8	  9074	     broker
    0	   110	  9148	   shared-process
    0	     0	  9372	     /bin/sh -c /bin/ps -ax -o pid=,ppid=,pcpu=,pmem=,command=
    0	     0	  9373	       /bin/ps -ax -o pid=,ppid=,pcpu=,pmem=,command=
    0	   205	  9206	   window (Form.js - world-countries - Visual Studio Code)
    0	    79	  9227	     extensionHost
    0	   229	  9257	       electron_node tsserver.js 
   16	    79	  9269	         electron_node typingsInstaller.js typesMap.js 
    0	    55	  9243	     watcherService

Workspace Stats: 
|  Window (Form.js - world-countries - Visual Studio Code)
|    Folder (world-countries): 55 files
|      File types: js(12) scss(10) json(6) map(5) html(2) txt(2) css(2)
|                  gitignore(1) md(1) LICENSE(1)
|      Conf files: package.json(1) settings.json(1)
</pre>

> Also, does this reproduce in the latest VS Code insiders build?
I didn't try, I'll now and get back to you asap.

**Comment by MohamedWagih96** [comment]
> Also, does this reproduce in the latest VS Code insiders build?

I tried it and it slightly better but the same problem remains. Moreover I think my OS become sluggish as soon as I launched VSCode Insider.
Here's the output of `code-insiders --status`

<pre>[11154:0220/134910.675787:ERROR:sandbox_linux.cc(369)] InitializeSandbox() called with multiple threads in process gpu-process.
<font color="#EF2929">[main 2020-02-20T11:49:10.699Z]</font> [File Watcher (node.js)] Failed to watch /home/rogue/.config/Code - Insiders/User for changes using fs.watch() (Error: ENOSPC: System limit for number of file watchers reached, watch &apos;/home/rogue/.config/Code - Insiders/User&apos;)
Version:          Code - Insiders 1.43.0-insider (9c341b3d8f2f41277be6dd59e7f4e847e6b594ac, 2020-02-20T05:39:45.923Z)
OS Version:       Linux x64 4.15.0-88-generic
CPUs:             Intel(R) Core(TM) i7-5500U CPU @ 2.40GHz (4 x 2890)
Memory (System):  7.71GB (1.13GB free)
Load (avg):       2, 1, 1
VM:               0%
Screen Reader:    no
Process Argv:     --no-sandbox --unity-launch
GPU Status:       2d_canvas:                     enabled
                  flash_3d:                      enabled
                  flash_stage3d:                 enabled
                  flash_stage3d_baseline:        enabled
                  gpu_compositing:               enabled
                  multiple_raster_threads:       enabled_on
                  oop_rasterization:             disabled_off
                  protected_video_decode:        unavailable_off
                  rasterization:                 disabled_software
                  skia_renderer:                 disabled_off_ok
                  video_decode:                  unavailable_off
                  viz_display_compositor:        enabled_on
                  viz_hit_test_surface_layer:    disabled_off_ok
                  webgl:                         enabled
                  webgl2:                        enabled

CPU %	Mem MB	   PID	Process
    0	   118	 10202	code-insiders main
    0	    32	 10205	   zygote
    0	    95	 10226	   gpu-process
    0	    47	 10242	   utility
    0	   103	 10317	   shared-process
    0	     0	 11177	     /bin/sh -c /bin/ps -ax -o pid=,ppid=,pcpu=,pmem=,command=
    0	     0	 11178	       /bin/ps -ax -o pid=,ppid=,pcpu=,pmem=,command=
    0	   245	 10575	   window (● WorldCountriesPage.js - world-countries - Visual Studio Code - Insiders)
    0	    95	 10597	     extensionHost
    0	    87	 10674	       electron_node tsserver.js 
    0	   331	 10675	       electron_node tsserver.js 
    7	   126	 10693	         electron_node typingsInstaller.js typesMap.js 
    0	    63	 10611	     watcherService

Workspace Stats: 
|  Window (● WorldCountriesPage.js - world-countries - Visual Studio Code - Insiders)
|    Folder (world-countries): 55 files
|      File types: js(12) scss(10) json(6) map(5) html(2) txt(2) css(2)
|                  gitignore(1) md(1) LICENSE(1)
|      Conf files: package.json(1) settings.json(1)
</pre>

**Comment by mjbvz** [comment]
@MohamedWagih96 Is it the `typingsInstaller.js` process that always has high cpu usage?

Also, does high cpu usage remains even when you stop doing anything in the editor for a minute? Some cpu usage is expected while editing files but it should die down once editing stops

**Comment by MohamedWagih96** [comment]
@mjbvz Yes it is always high even while i'm not doing any coding. Matter of fact the results I posted were when I just opened the laptop and nothing else was running except VSCode.

**Comment by mjbvz** [comment]
Is it always the `typingsInstaller.js` process?

**Comment by MohamedWagih96** [comment]
Yes
